Source/NSURL.m

```diff
@@ -1539,6 +1539,29 @@ - (NSString*) lastPathComponent
   return [[self path] lastPathComponent];
 }
 
+- (BOOL) isFileReferenceURL
+{
+  return NO;
+}
+
+- (NSURL *) fileReferenceURL
+{
+  if ([self isFileURL]) 
```

Comment:
The check here should be for isFileReferenceURL. For isFileURL a new reference should be created and currently we don't have any means to do so.

Reply:
You’re of course right in terms of strictly copying Apple APIs. However, given that we cannot reproduce this exact functionality and file reference URLs are just "special URLs", my thought was that returning self in case of file URLs would be better in terms of API compatibility, as clients wouldn’t expect this method to return nil for a file URL. I’m open to your thoughts tho.